src/domain/profile/Profile.tsx

```diff
@@ -48,6 +52,22 @@ const Profile: FunctionComponent = () => {
               <h1>
                 {data.myProfile.firstName} {data.myProfile.lastName}
               </h1>
+              <Button
+                ariaLabel={t('profile.edit.label')}
+                onClick={() => setIsOpen(true)}
+              >
+                <Icon src={settingsIcon} alt={t('profile.edit.label')} />
+              </Button>
+              {isOpen && (
+                <EditProfileModal
+                  onSubmit={() => {
+                    console.log('submit');
+                  }}
+                  isOpen={isOpen}
+                  setIsOpen={setIsOpen}
+                  initialValues={data.myProfile}
```

Comment:
put a conditional render, if (data.myProfile) is null just dont render

Reply:
But but if data.myProfile is empty you'll be redirected to the home page (see above the render function). Do we need to check it here too?

Reply:
ah true